BGL/include/CGAL/boost/graph/properties_Surface_mesh.h

```diff
@@ -105,125 +105,166 @@ struct property_map<CGAL::Surface_mesh<Point>, boost::edge_weight_t >
   typedef CGAL::SM_edge_weight_pmap<Point> type;
   typedef CGAL::SM_edge_weight_pmap<Point> const_type;
 };
-
+}
+namespace CGAL{
 template <typename Point>
-typename property_map<CGAL::Surface_mesh<Point>, boost::edge_weight_t>::const_type
+typename boost::property_map<CGAL::Surface_mesh<Point>, boost::edge_weight_t>::const_type
 get(boost::edge_weight_t, const CGAL::Surface_mesh<Point>& sm)
 {
   return CGAL::SM_edge_weight_pmap<Point>(sm);
 }
 
+// forward declarations, see <CGAL/Surface_mesh.h>
+class SM_Vertex_index;
+class SM_Edge_index;
+class SM_Halfedge_index;
+class SM_Face_index;
+
 template <typename Point>
-typename CGAL::SM_edge_weight_pmap<Point>::value_type
-get(boost::edge_weight_t, const CGAL::Surface_mesh<Point>& sm, 
-    const typename boost::graph_traits<CGAL::Surface_mesh<Point> >::edge_descriptor& e)
+typename CGAL::Kernel_traits<Point>::type::FT
```

Comment:
This highlights the fact that if one wants to use the default edge weight property map, one needs to use `Surface_mesh` with a CGAL Point (which is not documented).
cc @afabri